PaymentInitiation/Program.cs

```diff
@@ -389,16 +478,21 @@ private static HttpClient CreateGenericAuthClient()
         //
         // Create a http client with the basic common attributes set for a request to API:s
         //
-        private static HttpClient CreateGenericApiClient(string bicFi)
+        private static HttpClient CreateGenericApiClient(string bicFi, string psuIPAddress, string psuUserAgent, string psuCorporateId)
         {
             var apiClient = new HttpClient(_apiClientHandler);

```

Comment:
Not sure it matters in code that isn't mission critical, but it's generally advised not to create multiple instances of HttpClient. I suggest that you have two static private fields for each type of HttpClient that you need (auth, api), initialize them once and then set the X-Request-ID header on the HttpRequestMessage instead. Alternatively, you could have a helper method called GetApIClientWithUniqueHeader that would remove any existing X-Request-ID header and add a new one. This would however not be thread safe.